platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/capabilities/CapabilitiesConflictResolutionIntegrationTest.groovy

```diff
@@ -260,4 +257,75 @@ class CapabilitiesConflictResolutionIntegrationTest extends AbstractModuleDepend
             }
         }
     }
+
+    @RequiredFeature(feature = GradleMetadataResolveRunner.REPOSITORY_TYPE, value = "maven")
+    def "resolving a conflict does not depend on participant order"() {
+        given:
+        repository {
+            'org:testA:1.0' {
+                variant('runtime') {
+                    capability('org', 'testA', '1.0')
+                    capability('cap')
+                }
+            }
+            'org:testB:1.0' {
+                variant('runtime') {
+                    capability('org', 'testB', '1.0')
+                    capability('cap')
+                }
+            }
+            'org:testC:1.0' {
+                variant('runtime') {
+                    capability('org', 'testC', '1.0')
+                    capability('cap')
+                }
+            }
+        }
+
+        buildFile << """
+            dependencies {
+${deps.collect { "                conf(\"$it\")" }.join("\n")}
```

Comment:
💅 formatting looks a bit weird here

Reply:
That's "by design" so the produced file is formatted OK.

Reply:
Improved it